jupyterlab_chameleon/artifact.py

```diff
@@ -560,3 +608,63 @@ def get(self):
         except Exception as err:
             self.log.exception("An Unknown error occured")
             return self.error_response(500, str(err))
+
+
+class ArtifactMetricHandler(APIHandler, ErrorResponder):
+
+    def initialize(self, db: DB = None, notebook_dir: str = None):
+        self.api_client = ArtifactAPIClient(config=self.config)
+        self.db = db
+        self.notebook_dir = notebook_dir or "."
+
+    @web.authenticated
+    def put(self):
+        """Edit metric for an existing artifact."""
+        self.check_xsrf_cookie()
+
+        body = json.loads(self.request.body.decode("utf-8"))
+
+        uuid = None
+        version_slug = None
+        try:
+            local_contents = {
+                os.path.relpath(
+                    la.path, self.notebook_dir
+                ).replace("./", ""): (la.artifact_uuid, la.artifact_version_slug)
+                for la in self.db.list_artifacts()
+            }
+            artifact_path = body.pop("path", "")
+            if artifact_path in local_contents:
+                uuid = local_contents[artifact_path][0]
+                version_slug = local_contents[artifact_path][1]
+        except Exception as err:
+            self.log.exception("Unable to get artifact metadata")
+            return self.error_response(500, str(err))
+
+        try:
+            if uuid and version_slug:
+                metric_name = body.pop("metric")
+                if not metric_name:
+                    return self.error_response(400, "Missing metric name")
+
+                username = os.getenv("JUPYTERHUB_USER")
```

Comment:
I think we need to pull the user from a Keycloak token. The `JUPYTERHUB_USER` doesn't provide the same username as keycloak. For example, my Keycloak username is `adamcooper@uchicago.edu`, but my `JUPYTERHUB_USER` is `adamcooper`.